### PR TITLE
crimson: conditionally reload ObjectState if execution fails

### PR DESCRIPTION
**Pull request by rzarzynski**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by rzarzynski** [comment]
This PR is a continuation of #34257. It brings conditionality to the `obc` reloads. They will be skipped if `OpsExecuter` hasn't processed any write-like operation which should address the case of *compare-and-operate*.

File: src/crimson/osd/pg.h [hunk 1]
```diff
@@ -34,10 +34,10 @@
 #include "crimson/osd/pg_recovery_listener.h"
 #include "crimson/osd/recovery_backend.h"
 
-class OSDMap;
 class MQuery;
 class PGBackend;
 class PGPeeringEvent;
+class OSDMap;
 class osd_op_params_t;
```
Comment (rzarzynski) [inline src/crimson/osd/pg.h:41]: Oops, bump this up.

File: src/crimson/osd/pg.cc [hunk 2]
```diff
@@ -610,6 +610,25 @@ osd_op_params_t&& PG::fill_op_params_bump_pg_version(
     osd_op_p.user_at_version = osd_op_p.at_version.version;
   }
   return std::move(osd_op_p);
+
+seastar::future<Ref<MOSDOpReply>> PG::do_failed_op_processing(
```
Comment (tchaikov) [inline src/crimson/osd/pg.cc:614]: how about s/do_failed_op_processing/reply_failed_op/ or s/do_failed_op_processing/handle_failed_op/ ?

**Review by tchaikov — APPROVED** [review]

File: src/crimson/osd/pg.cc (hunk 2 above)
Comment (rzarzynski) [reply src/crimson/osd/pg.cc:614]: `reply_failed_op` would be too-focused and *replying* while we have even more important responsibility there. However, `handle_failed_op` is good! Taking it, thanks!

**Comment by tchaikov** [comment]
@rzarzynski ping?

@athanatos could you take a look at this change?

**Review by athanatos — APPROVED** [review]

**Pull request merged**